src/catalog/postgres/pg_proc_impl.cpp

```diff
@@ -0,0 +1,791 @@
+#include "catalog/postgres/pg_proc_impl.h"
+
+#include <vector>
+
+#include "catalog/database_catalog.h"
+#include "catalog/postgres/builder.h"
+#include "catalog/postgres/pg_namespace.h"
+#include "catalog/postgres/pg_proc.h"
+#include "common/error/error_code.h"
+#include "common/error/exception.h"
+#include "execution/functions/function_context.h"
+#include "storage/index/index.h"
+#include "storage/sql_table.h"
+#include "transaction/deferred_action_manager.h"
+
+namespace noisepage::catalog::postgres {
+
+PgProcImpl::PgProcImpl(db_oid_t db_oid) : db_oid_(db_oid) {}
+
+void PgProcImpl::BootstrapPRIs() {
+  const std::vector<col_oid_t> pg_proc_all_oids{PgProc::PG_PRO_ALL_COL_OIDS.cbegin(),
+                                                PgProc::PG_PRO_ALL_COL_OIDS.cend()};
+  pg_proc_all_cols_pri_ = procs_->InitializerForProjectedRow(pg_proc_all_oids);
+  pg_proc_all_cols_prm_ = procs_->ProjectionMapForOids(pg_proc_all_oids);
+
+  const std::vector<col_oid_t> set_pg_proc_ptr_oids{PgProc::PRO_CTX_PTR_COL_OID};
+  pg_proc_ptr_pri_ = procs_->InitializerForProjectedRow(set_pg_proc_ptr_oids);
+}
```

Comment:
Should this just be folded into `Bootstrap` (or per other comment, at least called from there)?

Reply:
Makes sense to me.